plugins/tiddlywiki/multiwikiserver/modules/mws-server.js

```diff
@@ -358,9 +359,75 @@ Server.prototype.isAuthorized = function(authorizationType,username) {
 	return principals.indexOf("(anon)") !== -1 || (username && (principals.indexOf("(authenticated)") !== -1 || principals.indexOf(username) !== -1));
 }
 
+Server.prototype.parseCookieString = function(cookieString) {
+	const cookies = {};
+	if (typeof cookieString !== 'string') return cookies;
+
+	cookieString.split(';').forEach(cookie => {
+			const parts = cookie.split('=');
+			if (parts.length >= 2) {
+					const key = parts[0].trim();
+					const value = parts.slice(1).join('=').trim();
+					cookies[key] = decodeURIComponent(value);
+			}
+	});
+
+	return cookies;
+}
+
+Server.prototype.authenticateUser = function(request, response) {
+	const {session: session_id} = this.parseCookieString(request.headers.cookie)
+	if (!session_id) {
+		return false;
+	}
+	// get user info
+	const user = this.sqlTiddlerDatabase.findUserBySessionId(session_id);
+	if (!user) {
+		return false
+	}
+	delete user.password;
+	// @TODO: implement logic to determine if a user is an admin user
+	user['isAdmin'] = true
+
+	return user
+};
+
+Server.prototype.requestAuthentication = function(response) {
+	if(!response.headersSent) {
+		response.writeHead(401, {
+			'WWW-Authenticate': 'Basic realm="Secure Area"'
+		});
+		response.end('Authentication required.');
+	}
+};
+
+Server.prototype.redirectToLogin = function(response, returnUrl) {
+	if (!response.headersSent) {
+			const validReturnUrlRegex = /^\/(?!.*\.(ico|png|jpg|jpeg|gif|svg|css|js|woff|woff2|ttf|eot|json)$).*$/;
```

Comment:
What is the purpose of this sanitisation?

Reply:
The url sanitization was done to prevent the user from being redirected to a file url instead of a page.